examples/tic-tac-toe/move/sources/owned.move

```diff
@@ -0,0 +1,326 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// An implementation of Tic Tac Toe, using owned objects.
+///
+/// The `Game` object is owned by an admin, so players cannot mutate the game
+/// board directly. Instead, they convey their intention to place a mark by
+/// transferring a `Mark` object to the `Game`.
+///
+/// This means that every move takes two owned object fast path operations --
+/// one by the player, and one by the admin. The admin could be a third party
+/// running a centralized service that monitors marker placement events and
+/// responds to them, or it could be a 1-of-2 multisig address shared between
+/// the two players, as demonstrated in the demo app.
+///
+/// The `shared` module shows a variant of this game implemented using shared
+/// objects, which provides different trade-offs: Using shared objects is more
+/// expensive, however the implementation is more straightforward and each move
+/// only requires one transaction.
+module tic_tac_toe::owned {
+    use sui::event;
+    use sui::transfer::Receiving;
+
+    // === Object Types ===
+
+    /// The state of an active game of tic-tac-toe.
+    public struct Game has key, store {
+        id: UID,
+
+        /// Marks on the board.
+        board: vector<u8>,
+
+        /// The next turn to be played.
+        turn: u8,
+
+        /// The address expected to send moves on behalf of X.
+        x: address,
+
+        /// The address expected to send moves on behalf of O.
+        o: address,
+
+        /// Public key of the admin address.
+        admin: vector<u8>,
+    }
+
+    /// The player that the next turn is expected from is given a `TurnCap`.
+    public struct TurnCap has key {
+        id: UID,
+        game: ID,
+    }
+
+    /// A request to make a play -- only the player with the `TurnCap` can
+    /// create and send `Mark`s.
+    public struct Mark has key, store {
+        id: UID,
+        player: address,
+        row: u8,
+        col: u8,
+    }
+
+    /// An NFT representing a finished game. Sent to the winning player if there
+    /// is one, or to both players in the case of a draw.
+    public struct Trophy has key {
+        id: UID,
+
+        /// Whether the game was won or drawn.
+        status: u8,
+
+        /// The state of the board at the end of the game.
+        board: vector<u8>,
+
+        /// The number of turns played
+        turn: u8,
+
+        /// The other player (relative to the player who owns this Trophy).
+        other: address,
+    }
+
+    // === Event Types ===
+
+    public struct MarkSent has copy, drop {
+        game: ID,
+        mark: ID,
+    }
+
+    public struct GameEnd has copy, drop {
+        game: ID,
+    }
+
+    // === Constants ===
+
+    // Marks
+    const MARK__: u8 = 0;
+    const MARK_X: u8 = 1;
+    const MARK_O: u8 = 2;
+
+    // Trophy status
+    const TROPHY_NONE: u8 = 0;
+    const TROPHY_DRAW: u8 = 1;
+    const TROPHY_WIN: u8 = 2;
```

Comment:
Should we wait a few weeks and just use enums?

Reply:
I don't think we should wait a few weeks to land the docs change, but we should come back and update it once enums is generally available, yes.

Reply:
Flagging for @tzakian